pkg/api-server/resource_endpoints.go

```diff
@@ -476,7 +478,9 @@ func (r *resourceEndpoints) updateResource(
 			log.Error(err, "Could not write the response")
 		}
 	} else {
-		response.WriteHeader(200)
+		if err := response.WriteHeaderAndJson(200, struct{}{}, "application/json"); err != nil {
```

Comment:
Should we return api_server_types.CreateOrUpdateSuccessResponse instead?

```go
response := api_server_types.CreateOrUpdateSuccessRespons{Warnings: model.Deprecations(currentRes)}
if err := response.WriteHeaderAndJson(200, response, "application/json"); err != nil {
	log.Error(err, "Could not write the response")
}
```

That's how it should have been written originally and we wouldn't have this issue at least :)